src/StructTypes.jl

```diff
@@ -132,6 +142,8 @@ isempty(::Number) = false
 isempty(::Nothing) = true
 isempty(x) = false
 isempty(x, i) = isempty(Core.getfield(x, i))
+isempty(::Type{T}, x) where {T} = isempty(x) # generic fallback
```

Comment:
Can you walk through why this is needed? What's your use-case?

Reply:
This allows users to customize the behavior of `isempty` for their own structs without modifying the behavior of `isempty` for other types or committing any type piracy.

For example, suppose that I want `missing` to be treated as empty when serializing my structs. I can just do `isempty(::Type{MyType}, ::Missing) = true`.

Reply:
A little more explanation is here: https://github.com/JuliaData/StructTypes.jl/pull/21